### ISSUE TITLE
fine tuning

### ISSUE THREAD

**Issue by VanitarNordic** [charlesq34/pointnet#28]
Hi

How to fine tune the model for different number of classes and a custom dataset?

**Comment by charlesq34** [comment]
Hi @VanitarNordic 

You asked a good question!
It requires extra attention in terms of finetuning a model with batch normalization. Unfortunately there isn't good official document yet.

Basically you need to:
1. define the network graph as that in train.py, you can add your own layers as well.
2. define a saver_restore = tf.train.Saver([variable list]) where the variable list contains all traininable variables and those in batch normalization layers. You can print tf.global_variables() to see what they are
3. initialize all variables with sess.run(init) as in train.py
4. saver_restore(sess, <path to checkpoint e.g. log/model.ckpt>), this step will load weights into variables

Then you can start training based on the restored model.

Here is a good reference in TensorFlow doc:
https://www.tensorflow.org/programmers_guide/variables

**Comment by charlesq34** [comment]
Let me know if you meet any specific questions. You are also welcomed to contribute.

**Comment by FrederikHaa** [comment]
Hello,

I can't seem to find any information about the encoding of the sem seg hdf5 files, which makes it hard to create a custom dataset.
Is the data format described anywhere?

This work looks very promising, but without the file description it is hard to apply it to my own application (semantic segmentation of LiDAR point clouds from outdoor scenes)

**Comment by FrederikHaa** [comment]
Nevermind me, I had not read the readme in the sem_seg folder.

For anyone wondering the original dataset can be found [here](http://buildingparser.stanford.edu/dataset.html).
It consists of raw colorized point clouds stored as plaintext.
Each row of the file corresponds to one point and includes the six values: x, y, z, r, g and b.

The text files can then be converted to the hdf5 file format used by the network using the collect_indoor3d_data.py and gen_indoor3d_h5.py scripts in the sem_seg folder.

This means that in order to generate your own dataset, all you need to do is to create your own plaintext files with one rgb point per line.

**Comment by charlesq34** [comment]
Hi @FrederikHaa 
sorry for the late response. It seems you have already figured it out!
